media-proxy/include/mesh/st2110rx.h

```diff
@@ -0,0 +1,174 @@
+#ifndef ST2110RX_H
+#define ST2110RX_H
+
+#include "st2110.h"
+
+namespace mesh::connection {
+
+/**
+ * ST2110Rx
+ *
+ * Base abstract class of ST2110Rx. ST2110_20Rx/ST2110_22Rx/ST2110_30Rx
+ * inherit this class.
+ */
+template <typename FRAME, typename HANDLE, typename OPS> class ST2110Rx : public ST2110 {
+  public:
+    ST2110Rx() : mtl_session(nullptr), ops{0}, transfer_size(0) { _kind = Kind::receiver; }
+
+    ~ST2110Rx() {
+        shutdown(_ctx);
+        if (ops.name)
+            free((void *)ops.name);
+    }
+
+  protected:
+    HANDLE mtl_session;
+    OPS ops;
+    size_t transfer_size;
+    std::jthread frame_thread_handle;
+    context::Context _ctx = context::WithCancel(context::Background());
+
+    virtual FRAME *get_frame(HANDLE) = 0;
+    virtual int put_frame(HANDLE, FRAME *) = 0;
+    virtual HANDLE create_session(mtl_handle, OPS *) = 0;
+    virtual int close_session(HANDLE) = 0;
+
+    int configure_common(context::Context& ctx, const std::string& dev_port,
+                         const MeshConfig_ST2110& cfg_st2110) {
+        int session_id = 0;
+        mtl_device = get_mtl_device(dev_port, MTL_LOG_LEVEL_CRIT, cfg_st2110.local_ip_addr, session_id);
+        if (mtl_device == nullptr) {
```

Comment:
Maybe this way?
```suggestion
        if (!mtl_device) {
```

Reply:
Fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/261/commits/92513fe839a25353cd6ab4ced946b0ab5ed474c9